syntax = "proto3";

service DeliveryService {
  rpc SendLocationUpdate (LocationRequest) returns (LocationResponse);
}

message LocationRequest {
  string vehicle_id = 1;
  double latitude = 2;
  double longitude = 3;
  string status = 4; // Delivery status: Pending, Pre-Transit, In Transit, etc.
}

message LocationResponse {
  string message = 1;
}
